content/en/admin/docker.md

```diff
@@ -0,0 +1,309 @@
+---
+
+We do not recommend admins use Docker Compose unless they have significant prior
+experience running docker in production. In our experience helping hundreds of
+admins troubleshoot and set up their server, there are many footguns and new
+areas of complexity compared to the standard systemd workflow that the Mastodon
+contributors are used to supporting.
+
+If that's something you're comfortable with, lets get started.
+
+## Setting up
+
+Clone Mastodon's repository.
+    
+    # Clone mastodon to ~/live directory
+    git clone https://github.com/tootsuite/mastodon.git live
+    # Change directory to ~/live
+    cd ~/live
+    # Checkout to the latest stable branch
+    git checkout $(git tag -l | grep -v 'rc[0-9]*$' | sort -V | tail -n 1)
+    
+
+Review the settings in `docker-compose.yml`. Note that it is **not default** to store the postgresql database and redis databases in a persistent storage location. If you plan on running your instance in production, you **must** uncomment the [`volumes` directive](https://github.com/tootsuite/mastodon/blob/972f6bc861affd9bc40181492833108f905a04b6/docker-compose.yml#L7-L16) in `docker-compose.yml`.
+
+## Getting the Mastodon image
+
+### Using a prebuilt image
+
+If you're not making any local code changes or customizations on your instance, you can use a prebuilt Docker image to avoid the time and resource consumption of a build. Images are available from Docker Hub: https://hub.docker.com/r/tootsuite/mastodon/
+    
+To use the prebuilt images:
+
+1. Open `docker-compose.yml` in your favorite text editor.
+   1. Comment out the `build: .` lines for all images (web, streaming, sidekiq).
+   2. Edit the `image: tootsuite/mastodon` lines for all images to include the release you want. The default is `latest` which is the most recent stable version, however it recommended to explicitly pin a version: If you wanted to use v2.2.0 for example, you would edit the lines to say: `image: tootsuite/mastodon:v2.2.0`
+   3. Save the file and exit the text editor.
+2. Run `cp .env.production.sample .env.production` to bootstrap the configuration. Edit the correct values now.
+3. Run `docker-compose build`. It will now pull the correct image from Docker Hub.
+4. Set correct file-owner with `sudo chown -R 991:991 public/system`
+
+### Building your own image
+
+You must build your own image if you've made any code modifications. To build your own image:
+
+1. Open `docker-compose.yml` in your favorite text editor.
+   1. Uncomment the `build: .` lines for all images (web, streaming, sidekiq) if needed.
+   2. Save the file and exit the text editor.
+2. Run `cp .env.production.sample .env.production` to bootstrap the configuration. Edit the correct values now.
+3. Run `docker-compose build`.
+4. Set correct file-owner with `chown -R 991:991 public`
+
+## Building the app
+
+Now the image can be used to generate a configuration with:
+
+    docker-compose run --rm web bundle exec rake mastodon:setup
```

Comment:
`docker compose`(docker-compose v2) also works

Reply:
Also, seems like db/redis of `.env.production` should have to changed before/while `docker-compose run --rm web bundle exec rake mastodon:setup` or adding admin account will be interrupted, and you have to reset password to access account(since random set password is not displayed)